src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unusedlocalvariable/InputUnusedLocalVariableNestedClasses4.java

```diff
@@ -0,0 +1,23 @@
+/*
+UnusedLocalVariable
+
+
+*/
+package com.puppycrawl.tools.checkstyle.checks.coding.unusedlocalvariable;
+
+public class InputUnusedLocalVariableNestedClasses4 {
+  int a = 12;
+
+  void foo() {
+    int a = 12; // violation, unused variable 'a'
+    int ab = 12; // violation, unused variable 'ab'
+
+    class asd {
+      Test a = new Test() {
```

Comment:
Add a test case where this assignment is one block deeper, like `if (...) { Test a = ....`

Reply:
@rnveach could you explain more about how to change this? I cannot directly change what you mentioned, since this is an inner class and I cannot directly put `if (....) { Test a = ...` there (must be in a function.)

Reply:
Sorry, I missed this specific case was at the class level instead of the method level.

I am looking at your code that talks about the `grandParent` level, and was wondering if adding another level broke something. What is the ast in this scenario? The `System.out.println`?

Reply:
No Worries! :) 

So basically, I am adding the following line
https://github.com/checkstyle/checkstyle/blob/15c18905aef6b87d280994607a0160b90e9806ef/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedLocalVariableCheck.java#L428
to also push the variable `a` in `Test a = new Test()` to the stack. In the old version of this check, such a variable will be directly ignored, so when popping the stack, the `a` variable in the outer space will be popped, and therefore, there was no violation of the var `a` in outer space.

Reply:
I am not sure how to add one more level in between in this case, but I think [InputUnusedLocalVariableNestedClasses4.java](https://github.com/checkstyle/checkstyle/blob/15c18905aef6b87d280994607a0160b90e9806ef/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unusedlocalvariable/InputUnusedLocalVariableNestedClasses4.java) has covered cases with multiple levels.

Reply:
This can be ignored for now.